src/main/cc/maze/cell/cell.h

```diff
@@ -3,7 +3,29 @@
 
 // Abstract class to represent a cell.
 public class Cell {
```

Comment:
If it is an abstract class then all of the methods have to be virtual and you dont have any implementation. Then other classes extend that class and provide the implementation. If you declare a class abstract then you shouldnt implement any methods really. Think of an abstract class like an interface in java. But we dont have interfaces in C++ so you have to just make classes with all virtual methods